test/functional/test-mustache.js

```diff
@@ -40,4 +40,32 @@ describe('Mustache', () => {
     expect(mustache.render('{{{value}}}', {value: '<b>abc</b>'})).to.equal(
         '<B>ABC</B>');
   });
+
+  it('should only expand own properties', () => {
+    const parent = {value: 'bc'};
+    const child = Object.create(parent);
+    const container = {parent: parent, child: child};
+    expect(mustache.render('a{{value}}', parent)).to.equal('abc');
+    expect(mustache.render('a{{value}}', child)).to.equal('a');
+    expect(mustache.render('a{{parent.value}}', container)).to.equal('abc');
+    expect(mustache.render('a{{child.value}}', container)).to.equal('a');
+  });
+
+  it('should NOT allow calls to builtin functions', () => {
+    // Calls on x.pop in classical Mustache would lead to builtin call and
+    // mutate on the 't' object. Here we will not allow it. We explicitly
+    // prohibit such calls.
+    const obj = {
+      't': {
+        '0': '0',
+        '1': '1',
+        'length': 2,
+        'x': []
+      }
+    };
+    expect(mustache.render(
+        '{{#t}}{{x.pop}}X{{x.pop}}{{/t}}' +
```

Comment:
Accessing nested attributes is a JS-specific extension to mustache. When you reference a nested function, it will be called with the current context as this, instead of the parent of the function.